.pre-commit-config.yaml

```
@@ -0,0 +1,21 @@
+    - id: trailing-whitespace
+    - id: end-of-file-fixer
+    - id: check-yaml
+
+- repo: https://github.com/pre-commit/mirrors-clang-format
+  rev: v16.0.0
+  hooks:
+    - id: clang-format
+      name: clang-format
+      files: "\\.(c|cc|cpp|cxx|h|hh|hpp|ipp|inl)$"
+      args: ["--style=file"]
+
```

YAMLlint (1.37.1)

[error] too many blank lines (1 > 0)

(empty-lines)